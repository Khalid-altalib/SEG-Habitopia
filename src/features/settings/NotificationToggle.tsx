import { Box, Switch, Text } from "native-base";
import React from "react";

type Props = {};

const NotificationToggle = (props: Props) => {
  return (
    <Box display="flex" justifyContent={"space-between"}>
      <Text>Recieve In-App Notifications </Text>
      <Switch size="md"></Switch>
    </Box>
  );
};

export default NotificationToggle;
