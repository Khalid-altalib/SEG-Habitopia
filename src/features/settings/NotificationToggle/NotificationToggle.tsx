import { Box, HStack, theme } from "native-base";
import React, { useState } from "react";
import { Switch } from "react-native";
import { useDispatch, useSelector } from "../../../app/hooks";
import { setSettings } from "../settingsSlice";
import Text from "@components/Text";
import { TextType } from "types";

type Props = {
  defaultValue: boolean;
};

/**
 * A component that toggles the associated setting state between
 * true and false
 *
 * @param props - The properties passed to the component.
 * @returns - The notification toggle component.
 */

const NotificationToggle = (props: Props) => {
  const [switchEnabled, setSwitchEnabled] = useState(props.defaultValue);
  const dispatch = useDispatch();

  const handleSwitchChange = () => {
    const invertedSwitchValue = !switchEnabled;
    setSwitchEnabled(invertedSwitchValue);
    dispatch(setSettings({ notifications: invertedSwitchValue }));
  };

  return (
    <HStack
      display={"flex"}
      justifyContent={"space-between"}
      alignItems="center"
      flexDirection="row"
      borderBottomWidth={1}
      _light={{ borderColor: "gray.300" }}
      _dark={{ borderColor: "blueGray.700" }}
      paddingBottom={3}
    >
      <Box>
        <Text type={TextType.Small}>Notifications </Text>
        <Text type={TextType.Subtle}>Choose to receive notifications</Text>
      </Box>
      <Box>
        <Switch
          value={switchEnabled}
          onChange={handleSwitchChange}
          trackColor={{ true: theme.colors.purple[700] }}
          thumbColor="white"
          testID="switch"
        />
      </Box>
    </HStack>
  );
};

export default NotificationToggle;
