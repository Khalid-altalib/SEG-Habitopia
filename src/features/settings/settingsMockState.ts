import { SettingsState } from "./settingsSlice";

/**
 * The mock state for the settings redux state
 */

const settingsMockState: SettingsState = {
  settings: {
    email: "test@test.com",
    password: "password",
    notifications: true,
    name: "Test User",
    biography: "Hello, I'm Test User.",
  },
  fetchSettings: {
    loading: false,
    error: "",
  },
  setSettings: {
    loading: false,
    error: "",
  },
};

export default settingsMockState;
