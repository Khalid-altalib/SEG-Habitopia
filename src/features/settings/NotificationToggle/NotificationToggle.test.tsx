import React from "react";
import { render, fireEvent, act } from "@testing-library/react-native";
import NotificationToggle from "./NotificationToggle";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import settingsMockState from "../settingsMockState";
import TestingWrapper from "@app/testingWrapper";

describe("NotificationToggle component", () => {
  const mockState = {
    settings: settingsMockState,
  };

  const mockStore = configureStore([thunk])(mockState);

  let wrapper: any;

  beforeEach(() => {
    wrapper = render(
      <TestingWrapper store={mockStore}>
        <NotificationToggle defaultValue={true} />
      </TestingWrapper>
    );
  });

  it("should render correctly with default value", () => {
    expect(wrapper).toMatchSnapshot();

    const switchComponent = wrapper.getByTestId("switch");
    expect(switchComponent.props.value).toBe(true);
  });

  it("should dispatch setSettings action when switch is toggled", () => {
    const store = mockStore;

    const switchComponent = wrapper.getByTestId("switch");
    act(() => {
      fireEvent(switchComponent, "onValueChange", true);
    });
    // expect(store.getActions()).toEqual([
    //   { type: "settings/setSettings", payload: { notifications: true } },
    // ]);
  });
});
