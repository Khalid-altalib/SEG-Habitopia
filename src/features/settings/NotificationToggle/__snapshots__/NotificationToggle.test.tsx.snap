// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationToggle component should render correctly with default value 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#d4d4d8",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 12,
      }
    }
  >
    <View
      dataSet={{}}
      style={{}}
    >
      <Text
        style={
          [
            {
              "color": "#18181b",
            },
            {
              "fontFamily": undefined,
              "fontSize": 15,
            },
            undefined,
          ]
        }
      >
        Notifications 
      </Text>
      <Text
        style={
          [
            {
              "color": "#18181b",
            },
            {
              "color": "#71717a",
              "fontFamily": undefined,
              "fontSize": 15,
            },
            undefined,
          ]
        }
      >
        Choose to receive notifications
      </Text>
    </View>
    <View
      dataSet={{}}
      style={{}}
    />
    <View
      dataSet={{}}
      style={{}}
    >
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#7e22ce"
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        testID="switch"
        thumbTintColor="white"
        value={true}
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
